src/evict/evict_page.c

```diff
@@ -692,6 +694,9 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
     } else if (!WT_SESSION_BTREE_SYNC(session))
         LF_SET(WT_REC_VISIBLE_ALL);
 
+    if (checkpoint_running)
+        LF_SET(WT_REC_CHECKPOINT_RUNNING);
```

Comment:
I think setting this flag here is correct but I want to make sure. The error case happens when eviction evicts pages from a btree and checkpoint is yet to get to the history store, and must have already processed this btree? Can the btree that eviction evicts be ahead of the checkpoint and if so would that be an error case or fine?

Reply:
Currently, we can't evict ahead of checkpoint. If we can, that is still run, because they use different snapshot. Eviction may see the updates checkpoint cannot see.

Reply:
Cool thanks.